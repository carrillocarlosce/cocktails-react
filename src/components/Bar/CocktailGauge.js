import React from "react";
export default () => <div>CocktailGauge</div>;
