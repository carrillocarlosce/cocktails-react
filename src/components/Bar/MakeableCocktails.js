import React from "react";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import { Link } from "react-router-dom";

const styles = theme => ({
  title: {
    fontSize: "1.5rem",
    margin: "1rem 0"
  },
  list: {
    maxHeight: "20rem",
    overflowY: "scroll"
  }
});

const MakeableCocktails = ({ makeableCocktails, classes }) => {
  return (
    <div>
      <Typography variant="h3" className={classes.title} gutterBottom>
        Makeable Cocktails
      </Typography>
      <Typography component="p" paragraph>
        These are the cocktails you can make...
      </Typography>
      <List className={classes.list}>
        {makeableCocktails.map(cocktail => (
          <ListItem
            button
            key={cocktail.name}
            component={Link}
            to={`/cocktails/${cocktail.slug}`}
          >
            <ListItemText primary={cocktail.name} />
          </ListItem>
        ))}
      </List>
    </div>
  );
};
export default withStyles(styles)(MakeableCocktails);
