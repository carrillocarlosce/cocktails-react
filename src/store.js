import appReducer from "./reducers";
import { createStore, applyMiddleware, compose } from "redux";
import { fetchCocktails, fetchIngredients } from "./services/cocktail.service";
import { loadCocktails, loadIngredients } from "./actions";
import thunk from "redux-thunk";

const devtools =
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();

const middlewares = compose(
  applyMiddleware(thunk),
  devtools || (a => a)
);

const store = createStore(appReducer, middlewares);

fetchCocktails().then(function(response) {
  store.dispatch(loadCocktails(response));
});

fetchIngredients().then(function(response) {
  store.dispatch(loadIngredients(response));
});

export default store;
