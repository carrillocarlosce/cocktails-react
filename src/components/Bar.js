import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import PopularIngredients from "./Bar/PopularIngredients";
import CocktailGauge from "./Bar/CocktailGauge";

import IngredientPicker from "./IngredientPicker";
import { bindActionCreators } from "redux";
import { setBar } from "../actions";
import { connect } from "react-redux";

const styles = theme => ({
  explanation: {
    marginBottom: "1em",
    padding: "1em 2em"
  },
  root: {
    ...theme.mixins.gutters,
    justifyContent: "center",
    padding: "2em 4em"
  }
});

const EditBar = ({ classes, bar, setBar }) => {
  return (
    <div className={classes.root}>
      <Paper className={classes.explanation}>
        <Typography color="inherit">
          <h2>Your Bar</h2>
        </Typography>
        <Typography color="inherit">
          <p>
            Currently these selections are not persisted (so if you refresh your
            page or leave and come back, you'll need to set them up again). What
            this gives you is the ability to say what you have in, then back on
            the cocktail browser you can set the filter to only show you things
            you can make.
          </p>
        </Typography>

        <Typography color="inherit">
          <p>
            <strong>Select the ingredients you have in your bar...</strong>
          </p>
        </Typography>

        <IngredientPicker
          selectedIngredients={bar}
          onIngredientsChange={selectedIngredients => {
            setBar(selectedIngredients);
          }}
        />
        <br />

        <Grid container className={classes.root} spacing={2}>
          <Grid item xs={6}>
            <CocktailGauge />
          </Grid>
          <Grid item xs={6}>
            <PopularIngredients />
          </Grid>
        </Grid>
      </Paper>
    </div>
  );
};

const mapStateToProps = state => ({
  bar: state.bar
});

const mapDispatchToProps = dispatch => ({
  setBar: bindActionCreators(setBar, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(EditBar));
