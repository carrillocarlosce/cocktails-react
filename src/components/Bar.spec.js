import React from "react";
import renderer from "react-test-renderer";
import Bar from "./Bar";
import store from "../store";
import { Provider } from "react-redux";

it("does not explode when rendered", () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Bar />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
