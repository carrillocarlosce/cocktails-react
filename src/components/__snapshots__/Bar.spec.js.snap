// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not explode when rendered 1`] = `
<div
  className="EditBar-root-2"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 EditBar-explanation-1 MuiPaper-rounded"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
    >
      Your Bar
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Select the ingredients you have in your bar...
    </p>
    <div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Absinthe
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Aperol
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Apricot brandy
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Blackberry liqueur
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Cachaca
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Calvados
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Campari
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Champagne
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Cherry liqueur
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Coconut milk
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Coffee liqueur
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Cognac
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Cola
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Cranberry juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Cream
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Créme de Cacao
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Créme de Cassis
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Créme de Menthe
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Cream liqueur
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Dark rum
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          DiSaronno
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          DOM Bénédictine
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Drambuie
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Dry White Wine
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Egg yolk
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Galliano
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Gin
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Ginger Ale
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Ginger beer
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Grapefruit juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Hot coffee
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Kirsch
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Lemon juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Lillet Blonde
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Lime juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Olive juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Orange Bitters
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Orange juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Peach puree
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Peach schnapps
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Pineapple juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Pisco
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Prosecco
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Raspberry liqueur
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Red Port
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Soda water
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Syrup
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Tequila
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Tomato juice
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Triple Sec
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Vermouth
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Vodka
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          Whiskey
        </span>
      </div>
      <div
        className="MuiChip-root IngredientPicker-chip-60 MuiChip-clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiChip-label"
        >
          White rum
        </span>
      </div>
    </div>
    <br />
    <div
      className="MuiGrid-root EditBar-root-2 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
      >
        <div>
          <h3
            className="MuiTypography-root MakeableCocktails-title-194 MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Makeable Cocktails
          </h3>
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
          >
            These are the cocktails you can make...
          </p>
          <ul
            className="MuiList-root MakeableCocktails-list-195 MuiList-padding"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
      >
        <div>
          <h3
            className="MuiTypography-root CocktailGauge-title-200 MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Cocktail Gauge
          </h3>
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
          >
            How many cocktails can you make with what's in your bar?
          </p>
          <svg
            className="CircularProgressbar CocktailGauge-progressBar-201"
            data-test-id="CircularProgressbar"
            viewBox="0 0 100 100"
          >
            <path
              className="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "stroke": "#fafafa",
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "0px",
                }
              }
            />
            <path
              className="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "stroke": "#3f51b5",
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "289.02652413026095px",
                }
              }
            />
            <text
              className="CircularProgressbar-text"
              style={
                Object {
                  "fill": "#3f51b5",
                }
              }
              x={50}
              y={50}
            >
              0
            </text>
          </svg>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
      >
        <div>
          <h3
            className="MuiTypography-root PopularIngredients-title-202 MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Popular Ingredients
          </h3>
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
          >
            These are popular ingredients not currently in your bar.
          </p>
          <table
            className="MuiTable-root"
          >
            <thead
              className="MuiTableHead-root"
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Ingredient
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Appearances
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
            >
              <tr
                className="MuiTableRow-root"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root PopularIngredients-cocktailNameContainer-203 MuiTableCell-body"
                  scope="row"
                >
                  <div>
                    <span>
                      Gin
                    </span>
                    <button
                      aria-label="Add"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </th>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  19
                </td>
              </tr>
              <tr
                className="MuiTableRow-root"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root PopularIngredients-cocktailNameContainer-203 MuiTableCell-body"
                  scope="row"
                >
                  <div>
                    <span>
                      Lemon juice
                    </span>
                    <button
                      aria-label="Add"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </th>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  18
                </td>
              </tr>
              <tr
                className="MuiTableRow-root"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root PopularIngredients-cocktailNameContainer-203 MuiTableCell-body"
                  scope="row"
                >
                  <div>
                    <span>
                      Vodka
                    </span>
                    <button
                      aria-label="Add"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </th>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  17
                </td>
              </tr>
              <tr
                className="MuiTableRow-root"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root PopularIngredients-cocktailNameContainer-203 MuiTableCell-body"
                  scope="row"
                >
                  <div>
                    <span>
                      Syrup
                    </span>
                    <button
                      aria-label="Add"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </th>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  17
                </td>
              </tr>
              <tr
                className="MuiTableRow-root"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root PopularIngredients-cocktailNameContainer-203 MuiTableCell-body"
                  scope="row"
                >
                  <div>
                    <span>
                      Lime juice
                    </span>
                    <button
                      aria-label="Add"
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </th>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  15
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
